modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/queue.scala

```diff
@@ -0,0 +1,106 @@
+package edu.gemini.seqexec.engine
+
+import scalaz._
+import Scalaz._
+
+/**
+  * A list of `Sequence`s.
+  */
+case class Queue[+A](sequences: List[Sequence[A]])
```

Comment:
I spent one morning pulling my hair about why I couldn't make `Queue`/`Sequence`/`Step` *Functor* instances when all types seem fine.  I still don't understand how Scala subtyping exactly works but it turns out, that in order to make it type check, I had to make that type parameter *covariant*.